lms/services/dashboard.py

```diff
@@ -88,6 +90,26 @@ def delete_dashboard_admin(self, dashboard_admin_id: int) -> None:
         """Delete an existing dashboard admin."""
         self._db.query(DashboardAdmin).filter_by(id=dashboard_admin_id).delete()
 
+    def get_request_admin_organizations(self, request) -> list[Organization]:
+        """Get the organization the current user is an admin in."""
+        if request.has_permission(Permissions.STAFF) and (
+            request_public_id := request.params.get("public_id")
+        ):
+            # We special case permissions/filtering for staff members
+            # If the requests contains a filter for one organization will act as if the staff member
+            # is an admin in that organization. That will give access to its data and filter by it
+            organization = self._organization_service.get_by_public_id(
+                request_public_id
+            )
+            if not organization:
+                raise HTTPNotFound()
```

Comment:
Considering the `public_id` will be provided in the query string, maybe we should very clearly specify what has not been found here.

I assume this will end up being rendered as an error template. Can we provide extra information to be displayed there, via the raised error?

Reply:
I think we should aim to provide better 404 error messages in general but for now as this will only be provided by the admin pages a 404 here indicates an error somewhere else.